app/presenters/content_embed_presenter.rb

```diff
@@ -0,0 +1,54 @@
+module Presenters
+  class ContentEmbedPresenter
+    def initialize(edition)
+      @edition = edition
+    end
+
+    def render_embedded_content(details)
+      return details if details[:body].nil?
+
+      details[:body] = if details[:body].is_a?(Array)
+                         details[:body].map do |content|
+                           {
+                             content_type: content[:content_type],
+                             content: render_embedded_editions(content[:content]),
+                           }
+                         end
+                       else
+                         render_embedded_editions(details[:body])
+                       end
+
+      details
+    end
+
+  private
+
+    def render_embedded_editions(content)
+      embedded_content_references = EmbeddedContentFinderService.new.find_content_references(content)
+      embedded_content_references_by_content_id = embedded_content_references.index_by(&:content_id)
+
+      target_content_ids = @edition
+        .links
+        .where(link_type: "embed")
+        .map(&:target_content_id)
```

Comment:
nit: I think you could do 
```suggestion
        .pluck(:target_content_id)
```

which would avoid allocating Link objects for each link and then calling `#target_content_id` on each one (instead just returning the content ids directly from the SQL query.

Reply:
Nice, will update this.